scripts/build/autotools/CheckSSL.m4

```diff
@@ -0,0 +1,240 @@
+      PHP_ADD_LIBRARY([ssl],,[MONGODB_SHARED_LIBADD])
+      PHP_ADD_LIBRARY([crypto],,[MONGODB_SHARED_LIBADD])
+
+      if test -n "$OPENSSL_LIBDIR"; then
+        PHP_ADD_LIBPATH([$OPENSSL_LIBDIR],[MONGODB_SHARED_LIBADD])
+      fi
+
+      if test -n "$OPENSSL_INCDIR"; then
+        PHP_ADD_INCLUDE($OPENSSL_INCDIR)
+      fi
+
+      old_CFLAGS="$CFLAGS"
+      CFLAGS="$OPENSSL_INCDIR $CFLAGS"
+
+      AC_CHECK_DECLS([ASN1_STRING_get0_data],
+                     [have_ASN1_STRING_get0_data="yes"],
+                     [have_ASN1_STRING_get0_data="no"],
+                     [[#include <openssl/asn1.h>]])
+
+      CFLAGS="$old_CFLAGS"
+
+      PHP_MONGODB_SSL="openssl"
+      found_openssl="yes"
+    fi
+  fi
+
+  if test "$PHP_MONGODB_SSL" = "openssl" -a "$found_openssl" != "yes"; then
+    AC_MSG_ERROR([OpenSSL libraries and development headers could not be found])
+  fi
+])
+
+AS_IF([test "$PHP_MONGODB_SSL" = "libressl" -o "$PHP_MONGODB_SSL" = "auto"],[
+  found_libressl="no"
+
+  dnl TODO: this may require adding libcrypto to the modules list, as pkg-config does not seem to include it (unlike openssl)
+  PKG_CHECK_MODULES([PHP_MONGODB_SSL],[libtls],[
+    PHP_EVAL_INCLINE([$PHP_MONGODB_SSL_CFLAGS])
+    PHP_EVAL_LIBLINE([$PHP_MONGODB_SSL_LIBS],[MONGODB_SHARED_LIBADD])
+    PHP_MONGODB_SSL="libressl"
+    found_libressl="yes"
+  ])
+
+  if test "$found_libressl" = "no"; then
+    PHP_CHECK_LIBRARY([crypto],
+                      [EVP_DigestInit_ex],
+                      [have_crypto_lib="yes"],
+                      [have_crypto_lib="no"])
+    PHP_CHECK_LIBRARY([tls],
+                      [tls_init],
+                      [have_ssl_lib="yes"],
+                      [have_ssl_lib="no"],
+                      [-lcrypto])
+
+    if test "$have_ssl_lib" = "yes" -a "$have_crypto_lib" = "yes"; then
+      PHP_ADD_LIBRARY([tls],,[MONGODB_SHARED_LIBADD])
+      PHP_ADD_LIBRARY([crypto],,[MONGODB_SHARED_LIBADD])
+      PHP_MONGODB_SSL="libressl"
+      found_libressl="yes"
+    fi
+  fi
+
+  if test "$PHP_MONGODB_SSL" = "libressl" -a "$found_libressl" != "yes"; then
+    AC_MSG_ERROR([LibreSSL libraries and development headers could not be found])
+  fi
+])
+
+AS_IF([test "$PHP_MONGODB_SSL" = "darwin" -o \( "$PHP_MONGODB_SSL" = "auto" -a "$os_darwin" = "yes" \)],[
+  dnl PHP_FRAMEWORKS is only used for SAPI builds, so use MONGODB_SHARED_LIBADD for shared builds
+  if test "$ext_shared" = "yes"; then
+    MONGODB_SHARED_LIBADD="-framework Security -framework CoreFoundation $MONGODB_SHARED_LIBADD"
+  else
+    PHP_ADD_FRAMEWORK([Security])
+    PHP_ADD_FRAMEWORK([CoreFoundation])
+  fi
+  PHP_MONGODB_SSL="darwin"
+])
+
+AS_IF([test "$PHP_MONGODB_SSL" = "auto"],[
+  PHP_MONGODB_SSL="no"
+])
+
+AC_MSG_CHECKING([which TLS library to use])
+AC_MSG_RESULT([$PHP_MONGODB_SSL])
+
+dnl Disable Windows SSL and crypto
+AC_SUBST(MONGOC_ENABLE_SSL_SECURE_CHANNEL, 0)
+AC_SUBST(MONGOC_ENABLE_CRYPTO_CNG, 0)
```

Comment:
Is the idea to support these later?

Reply:
I don't believe so, since Windows builds use `config.w32` and will never use Autotools. We do still need to make the substitutions, though.